app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt

```diff
@@ -85,9 +87,62 @@ class FCMManagerImpl(
         }
     }
 
-    override fun showNotification(payload: NotificationPayload) {
-        Timber.d("showNotification: ${payload.conversationSid}")
+    fun createBitmapFromVectorDrawable(drawableId: Int): Bitmap? {
+        val drawable = ContextCompat.getDrawable(context, drawableId)
+        if (drawable == null) {
+            return null
+        }
+
+        val bitmap = Bitmap.createBitmap(200, 200, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+        drawable.setBounds(0, 0, canvas.getWidth(), canvas.getHeight())
+        drawable.draw(canvas)
+        return bitmap
+    }
 
+    val NotificationPayload.largeIconId: Int get() = when (type) {
+        NotificationPayload.Type.NEW_MESSAGE -> when {
+            mediaCount > 1 -> R.drawable.ic_media_multiple_attachments
+            mediaCount > 0 -> with(mediaContentType) {
+                when {
+                    startsWith("image/") -> R.drawable.ic_media_image
+                    startsWith("video/") -> R.drawable.ic_media_video
+                    startsWith("audio/") -> R.drawable.ic_media_audio
+                    else -> R.drawable.ic_media_document
+                }
+            }
+            else -> R.drawable.ic_media_document
+        }
+        else -> R.drawable.ic_media_document
+    }
+
+    val NotificationPayload.largeIcon get() = largeIconId?.let { createBitmapFromVectorDrawable(it) }
+
+    fun getContentTextForNotification(payload: NotificationPayload): String {
```

Comment:
plz make it `NotificationPayload.textForNotification` - the same way as `NotificationPayload.largeIconId` above. Readability much better...

Reply:
fixed